plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueToTrinoConverter.java

```diff
@@ -136,7 +136,17 @@ private static Column convertColumn(com.amazonaws.services.glue.model.Column glu
             //TODO(https://github.com/trinodb/trino/issues/7240) Add tests
             return new Column(glueColumn.getName(), HiveType.HIVE_STRING, Optional.ofNullable(glueColumn.getComment()));
         }
-        return new Column(glueColumn.getName(), HiveType.valueOf(glueColumn.getType().toLowerCase(Locale.ENGLISH)), Optional.ofNullable(glueColumn.getComment()));
+        return new Column(glueColumn.getName(), convertType(glueColumn), Optional.ofNullable(glueColumn.getComment()));
+    }
+
+    private static HiveType convertType(com.amazonaws.services.glue.model.Column column)
+    {
+        try {
+            return HiveType.valueOf(column.getType().toLowerCase(Locale.ENGLISH));
+        }
+        catch (IllegalArgumentException e) {
+            throw new TrinoException(HIVE_INVALID_METADATA, "Glue column '%s' has invalid data type: %s".formatted(column.getName(), column.getType()));
```

Comment:
would column name itself be sufficient to identify the offending table?

Reply:
Maybe not, but it's better than the `IllegalArgumentException` you get today.

Reply:
I updated this to include the table name